### PR TITLE
Fixed issue #19038: Privilege escalation bug to creation survey-group with others group as parent

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/models/services/SurveysGroupCreator.php [hunk 1]
```diff
@@ -51,7 +55,24 @@ public function __construct(
...
     public function save()
     {
-        $this->surveysGroup->attributes = $this->request->getPost('SurveysGroups');
+        $attributes = $this->request->getPost('SurveysGroups');
+
+        // Check if the parent group is valid
+        if (!empty($attributes['parent_id'])) {
+            $parentId = $attributes['parent_id'] ;
+            /* Check permission */
+            $availableParents = $this->surveysGroup->getParentGroupOptions();
```
Comment (Shnoulle) [inline application/models/services/SurveysGroupCreator.php:64]: The getParentGroupOptions have more code than https://github.com/LimeSurvey/LimeSurvey/blob/76bb242b4224d0ae58f11a50d6c1f6adac58d4fd/application/models/SurveysGroups.php#L347

(we still need a scope for permission … )

**Review by Shnoulle — COMMENTED** [review]
Some comment

1. getSurveyGroupsList is already used for Survey 
2. Add it to rules (and same way than Surveys maybe ?)

**Comment by Shnoulle** [comment]
_General comment for discussion_ : We don't have a permission about "Create child group in this survey group", maybe a feature here ?
Maybe it's more clear …
Here : allow to see => allow to use , maybe need «Permission to modify survey group security settings» ?

Capture d’écran du 2023-09-06 09-53-11

**Comment by gabrieljenik** [comment]
Got lost in this. Can you re-review @Shnoulle  ? Thanks

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**